SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -54,7 +52,69 @@ int GenerateCompleteSchemaFromClass(FString SchemaPath, int ComponentId, UClass*
 	return NumComponents;
 }
 
-bool CheckClassNameListValidity(const TArray<UClass*>& Classes)
+bool CheckIdentifierNameValidity(TSharedPtr<FUnrealType> TypeInfo)
+{
+	// Check Replicated Data
+	FUnrealFlatRepData RepData = GetFlatRepData(TypeInfo);
+	for (EReplicatedPropertyGroup Group : GetAllReplicatedPropertyGroups())
+	{
+		TMap<FString, TSharedPtr<FUnrealProperty>> SchemaReplicatedDataNames;
+		for (auto& RepProp : RepData[Group])
+		{
+			FString NextSchemaReplicatedDataName = SchemaFieldName(RepProp.Value);
+			TSharedPtr<FUnrealProperty>* ExistingReplicatedProperty = SchemaReplicatedDataNames.Find(NextSchemaReplicatedDataName);
+
+			if (ExistingReplicatedProperty != nullptr)
+			{
+				UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Replicated property name collision after removing non-alphanumeric characters: '%s' and '%s' - schema not generated"), *ExistingReplicatedProperty->Get()->Property->GetName(), *RepProp.Value->Property->GetName());
+				return false;
+			}
+
+			SchemaReplicatedDataNames.Add(NextSchemaReplicatedDataName, RepProp.Value);
+		}
+	}
+
+	// Check Handover data
+	FCmdHandlePropertyMap HandoverData = GetFlatHandoverData(TypeInfo);
+	TMap<FString, TSharedPtr<FUnrealProperty>> SchemaHandoverDataNames;
+	for (auto& Prop : HandoverData)
+	{
+		FString NextSchemaHandoverDataName = SchemaFieldName(Prop.Value);
```

Comment:
This code could be lifted out to a templated helper function to be used in both replicated properties and handover. might convolute the code but would reduce the size of the function. @joshuahuburn what do you think?

Reply:
Yeah it's surely possible if we get some refactoring time :D 
At the moment I wouldn't consider it a priority though, we should focus on enabling schema names with 'bad characters' rather than detecting them and not allowing them.